### PR TITLE
[Backport 2.x] Added support of timestamp/date/time using curly brackets

### PR DESCRIPTION
**Pull request by matthewryanwells**
Backport https://github.com/opensearch-project/sql/commit/1a7134b79044de655445f2c1e24ec56467544f50 from https://github.com/opensearch-project/sql/pull/1894

### CODE REVIEW

**Comment by matthewryanwells** [comment]
> Something wrong with doctests @matthewryanwells

This was due to a bad cherry pick I had done, it is fixed now

**Review by GumpacG — APPROVED** [review]

**Review by MitchellGale — APPROVED** [review]

File: sql/src/test/java/org/opensearch/sql/sql/antlr/SQLParserTest.java [hunk 1]
```diff
@@ -1,3 +1,9 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
```
Comment (acarbonetto) [inline sql/src/test/java/org/opensearch/sql/sql/antlr/SQLParserTest.java:4]: this seems unrelated, btw.  It's done on purpose, right?

**Review by Yury-Fridlyand — APPROVED** [review]

File: sql/src/test/java/org/opensearch/sql/sql/antlr/SQLParserTest.java (hunk 1 above)
Comment (matthewryanwells) [reply sql/src/test/java/org/opensearch/sql/sql/antlr/SQLParserTest.java:4]: It was done on purpose, I do realize that is unrelated and I probably should have made a separate PR for it, but I noticed it was missing the header when I was adding the header to the file I had created.

**Review by Yury-Fridlyand — APPROVED** [review]

**Comment by matthewryanwells** [comment]
After rebasing my code again with the most recent changes it has fixed the code coverage issues
